### ISSUE TITLE
Blockquote note and warning support

### ISSUE THREAD

**Issue by xt0rted** [mjbvz/vscode-github-markdown-preview#23]
GitHub Flavored Markdown now supports special styling for note and warning blockquotes. https://github.com/community/community/discussions/16925

> **Note**
> This is a note

> **Warning**
> This is a warning

**Comment by jjspace** [comment]
Just as a note this was a poorly thought out and implemented feature that's no longer documented and could go away at any moment. There's a lot of discussion in that linked thread but many people agree it should not have been added the way it was so I'm not sure it's worth trying to support for this extension. Specifically see [this comment thread](https://github.com/orgs/community/discussions/16925#discussioncomment-2791869)

**Comment by adam-grant-hendry** [comment]
It's been a minute and the [GitHub Docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) now state the beta testing syntax is deprecated and the official syntax is now:

```markdown
> [!NOTE]
> Highlights information that users should take into account, even when skimming.

> [!IMPORTANT]
> Crucial information necessary for users to succeed.

> [!WARNING]
> Critical content demanding immediate user attention due to potential risks.
```

Any chance we can reconsider supporting the syntax?

**Comment by ShadiestGoat** [comment]
I would love this to be added! 

As a side note, perhaps this issue should be renamed to include something like "alerts" or "admonitions"?

**Comment by to-sta** [comment]
Would also like to see this added 😄

**Comment by ByPikod** [comment]
@ShadiestGoat @adam-grant-hendry @xt0rted 

I was looking for this feature and found this repository. I thought I could contribute to this repository by introducing the desired feature. However, I soon realized that the "blockquote note syntax" couldn't be implemented using only CSS within this repository.

As a solution, I developed a MarkdownIt plugin to convert Markdown into HTML. Then, I created a VS Code extension that utilizes this plugin along with a CSS file I wrote. You can find the extension here:
https://marketplace.visualstudio.com/items?itemName=yahyabatulu.vscode-markdown-alert

**Comment by uncenter** [comment]
Thank you @ByPikod! Works fabulously.

**Comment by cyfyifanchen** [comment]
thank you. @ByPikod

**Comment by GuySartorelli** [comment]
As mentioned in https://github.com/mjbvz/vscode-github-markdown-preview/issues/23#issuecomment-1781285505 this can't be fully implemented with CSS alone.
It may be possible to at least change the colour of the left border of the block quote to match the relevant alert, which would be better than nothing IMO.

**Comment by AMDphreak** [comment]
> GitHub Flavored Markdown now supports special styling for note and warning blockquotes. [community/community#16925](https://github.com/orgs/community/discussions/16925)

```
> 
> > **Note**
> > This is a note
> 
> > **Warning**
> > This is a warning
```

Those are not the GitHub supported syntaxes. The correct syntax uses brackets and exclamation and are all-caps:

> [!NOTE]
> note block

> [!WARNING]
> warning block

> [!TIP]
> tip block

> [!CAUTION]
> caution block

```
> [!NOTE]
> note block

> [!WARNING]
> warning block

> [!TIP]
> tip block

> [!CAUTION]
> caution block
```

**Comment by AMDphreak** [comment]
> [@ShadiestGoat](https://github.com/ShadiestGoat) [@adam-grant-hendry](https://github.com/adam-grant-hendry) [@xt0rted](https://github.com/xt0rted)
> 
> I was looking for this feature and found this repository. I thought I could contribute to this repository by introducing the desired feature. However, I soon realized that the "blockquote note syntax" couldn't be implemented using only CSS within this repository.
> 
> As a solution, I developed a MarkdownIt plugin to convert Markdown into HTML. Then, I created a VS Code extension that utilizes this plugin along with a CSS file I wrote. You can find the extension here: https://marketplace.visualstudio.com/items?itemName=yahyabatulu.vscode-markdown-alert

yeah well nobody's using it. This plugin ranks high on the plugin marketplace's search results, because most people don't care about the additional feature you added. It needs to be integrated. This kind of uncoordinated effort is why the plugin marketplace is such a pain in the butt to use. I don't want to sit and shop for better alternatives and extensions for my extensions for my extensions all day.

Do you see your plugin anywhere on here?
![Image](https://github.com/user-attachments/assets/3d5285bf-1cb6-40fb-a36c-81b37138b4e0)

**Comment by ejm554** [comment]
> As a solution, I developed a MarkdownIt plugin to convert Markdown into HTML. Then, I created a VS Code extension that utilizes this plugin along with a CSS file I wrote. You can find the extension here: https://marketplace.visualstudio.com/items?itemName=yahyabatulu.vscode-markdown-alert

This is a helpful stopgap solution, @ByPikod. Thanks for writing it! Still hoping to see alerts integrated into this Markdown Preview extension someday.

**Comment by AMDphreak** [comment]
> > As a solution, I developed a MarkdownIt plugin to convert Markdown into HTML. Then, I created a VS Code extension that utilizes this plugin along with a CSS file I wrote. You can find the extension here: https://marketplace.visualstudio.com/items?itemName=yahyabatulu.vscode-markdown-alert
> 
> This is a helpful stopgap solution, [@ByPikod](https://github.com/ByPikod). Thanks for writing it! Still hoping to see alerts integrated into this Markdown Preview extension someday.

That's cool. Can you upload it to the openVSX marketplace too? Nobody using Cursor can use your plugin, because Microsoft is mean to 3rd parties while calling itself a champion of open source philosophy (kekw)

**Comment by AMDphreak** [comment]
Why are you sending me this? Are you suggesting it is true, or asking me to
investigate?

Gemini: https://g.co/gemini/share/af307f146dff
[image: image.png]


Grok:
https://grok.com/share/bGVnYWN5LWNvcHk%3D_2a2477ad-46a0-4582-ba2a-f467683e6508

[image: image.png]


And, moreover, couldn't you just realize that this is totally false? There
is no overwhelming backlash from society about this. You're watching total
crap.

**Comment by AMDphreak** [comment]
When you come across these overwhelmingly clickbaity titles, go ask Grok.
Hell, buy a grok subscription. Grok is owned by Elon Musk, and his team
works hard to make sure their prompt exposes as much false information as
possible.

**Comment by AMDphreak** [comment]
Elon Musk is obsessed with the phrase "maximum truth seeking" and wants his
team to bake that philosophy into Grok.

**Comment by dmillar-vx** [comment]
Appears to be fixed now, possibly as part of changes in https://github.com/mjbvz/vscode-github-markdown-preview-style
